File: 백준/Silver/1049. 기타줄/기타줄.py
N , M = map(int, input().split())
pack = [] 
ones = [] 
for _  in range(M):
    p , o = map(int, input().split())
    pack.append(p)
    ones.append(o) 


pac = min(pack)
one = min(ones) 


print( min( pac*(N//6)+pac , pac*(N//6)+ ( N%6)*one , N*one)) 

File: 백준/Silver/1049. 기타줄/기타줄.cc

#include <iostream>
#include <vector> 
#include <string> 
#include <algorithm> 
using namespace std; 

int main() { 
    int N ,M ; 
    cin >> N >> M ; 
    
    vector<int> packs;
    vector<int> ones;
    int p , o ; 
    for( int a = 0 ; a <M ; a++){
        cin >> p >> o ; 
        packs.push_back(p);
        ones.push_back(o); } 

    int pack = *min_element(packs.begin() , packs.end() ) ; 
    int one = *min_element(ones.begin() , ones.end() ) ;
    
    cout << min( { ( N/6)*pack+ pack , ( N/6)*pack + (N%6)*one  , one*N}) ;

} 

